src/Dialect/ONNX/ONNXOps/Math/Bernoulli.cpp

```diff
@@ -26,6 +26,11 @@ using namespace onnx_mlir;
 // Shape Inference
 //===----------------------------------------------------------------------===//
 
+ONNXOpShapeHelper *ONNXBernoulliOp::getShapeHelper(Operation *op,
+    ArrayRef<mlir::Value> oper, IndexExprBuilder *ieb, IndexExprScope *scope) {
+  return getNewShapeHelper<ONNXUnaryOpShapeHelper>(op, oper, ieb, scope);
+}
```

Comment:
It looks like this code is not automatically generated, doesn't it?

I think the current infrastructure you built is quite good so that we see repeated calling patterns for all operations. Given that, it looks to me, we can define a list of operations that need `ShapeHelperOpInterface` in `gen_onnx_mlir.py` and automatically generate `getShapeHelper` for each operation by using `gen_onnx_mlir.py`.

To do that, each operation need to have a shape helper with the same name, `ONNXBernoulliOpShapeHelper` for `ONNXBernoulliOp`. In such a case, a new class `ONNXBernoulliOpShapeHelper` that inherits `ONNXUnaryOpShapeHelper` is needed. What do you think?

Reply:
Yes, that is correct. There are many ops that reuse a shape type at the moment, for example all of the Unary and Binary. 

If you want me to go that route, I will have to put a 
```
using ONNXxxxOpShapeHelper = ONNXUnaryOpShapeHelper<ONNXxxxOp>
```
for each op that does not already have a specific helper for it. 

Let me know if you prefer that pattern, and I can try migrate to that pattern.

Reply:
Yes, please migrate to that pattern, so that we have a uniform interface for all ops' shape helper. Thanks!

Reply:
will do

Reply:
> The only way this could work if there was a way to define in the ODS of the ONNX op code to be only included in the cpp file, which I cannot find at this time.

As I understand, `extraClassDefinition` will add code to the created cpp file and `extraClassDeclaration` will add code to the created hpp file. So if we don't have `extraClassDeclaration`, the code will be only added to the cpp file. Does it still not work because of cyclic dependency?

Reply:
It looks like I can disable the interface for the ONNX Ops, and add the shapeHelper as `extraClassDeclaration` with the code once I remove the dependence from ONNXOps/ShapeHelper.hpp to ONNXOps.hpp, by using the forward declaration of all of the ONNX Ops classes found using GET_OP_FWD_CLASSES define in conjunction with the automatically generated ONNXOps.hpp.inc (which will define only the `class ONNXAddOp;` type fo refs).

Additional ONNX Ops have to be handled separately.